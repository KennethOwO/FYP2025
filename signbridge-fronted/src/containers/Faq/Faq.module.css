@import "@radix-ui/colors/black-alpha.css";
@import "@radix-ui/colors/green.css";
@import "@radix-ui/colors/mauve.css";
@import "@radix-ui/colors/violet.css";

.layout {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: var(--imageHeight);
}

.faqImage {
    width: 100%;
    height: auto;
    -webkit-box-reflect: below 0px -webkit-gradient(linear, left top, left bottom, from(transparent), color-stop(0.5, transparent), to(rgba(255, 255, 255, 0.25)));
    position: relative;
    z-index: 1;
}

.AccordionRoot {
    border-radius: 6px;
    width: 60%;
    margin: 50px auto;
    position: relative;
    z-index: 2;
}

.AccordionItem {
    position: relative;
    overflow: hidden;
    margin-top: 16px;
    margin-bottom: 30px;
    border-radius: 8px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    transition: box-shadow 0.3s ease-in-out;
}

.AccordionItem:first-child {
    margin-top: 40px;
}

.AccordionItem:last-child {
    margin-bottom: 0;
}

.AccordionItem:focus-within {
    box-shadow: 0 6px 12px rgba(0, 0, 0, 0.15);
}

.AccordionHeader {
    display: flex;
}

.AccordionTrigger {
    all: unset;
    font-family: inherit;
    background-color: transparent;
    padding: 5px 20px;
    height: 45px;
    flex: 1;
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: 17px;
    line-height: 1;
    color: #8d4e13;
    box-shadow: 0 1px 0 var(--mauve-6);
    background-color: #f8e2cd;
    transition: transform 0.3s ease-in-out;
}

.AccordionTrigger:hover {
    cursor: pointer;
}

.AccordionItem:hover {
    transform: scale(1.05);
    z-index: 1;
    transition: transform 0.3s ease-in-out;
}

.AccordionContent {
    position: relative;
    overflow: hidden;
    font-size: 16px;
    color: var(--mauve-11);
    background-color: var(--mauve-2);
}

.AccordionContent[data-state="open"] {
    animation: slideDown 300ms cubic-bezier(0.87, 0, 0.13, 1);
}

.AccordionContent[data-state="closed"] {
    animation: slideUp 300ms cubic-bezier(0.87, 0, 0.13, 1);
}

.AccordionContentText {
    padding: 15px 20px;
}

.AccordionChevron {
    color: black;
    transition: transform 300ms cubic-bezier(0.87, 0, 0.13, 1);
}

.AccordionTrigger[data-state="open"] > .AccordionChevron {
    transform: rotate(180deg);
}

@keyframes slideDown {
    from {
        height: 0;
    }
    to {
        height: var(--radix-accordion-content-height);
    }
}

@keyframes slideUp {
    from {
        height: var(--radix-accordion-content-height);
    }
    to {
        height: 0;
    }
}

@media screen and (max-width: 768px) {
    .AccordionRoot {
        width: 90%;
    }

    .AccordionTrigger {
        font-size: 0.9rem;
    }

    .AccordionContentText {
        font-size: 0.9rem;
    }
}

@media screen and (max-width: 480px) {
    .AccordionRoot {
        width: 90%;
    }

    .AccordionTrigger {
        font-size: 0.85rem;
    }

    .AccordionContentText {
        font-size: 0.85rem;
    }
}
