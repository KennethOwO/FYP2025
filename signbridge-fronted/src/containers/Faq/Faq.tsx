import React, { useState, useEffect } from "react";
import styles from "./Faq.module.css";
import * as Accordion from "@radix-ui/react-accordion";
import classNames from "classnames";
import { ChevronDownIcon } from "@radix-ui/react-icons";
import { FetchFaq } from "../../services/faq.service";
import { useTranslation } from "react-i18next";

interface FaqData {
  question_en: string;
  question_bm: string;
  answer_en: string;
  answer_bm: string;
  faq_id: number;
}

interface AccordionTriggerProps
  extends React.ComponentPropsWithoutRef<typeof Accordion.Trigger> {
  children: React.ReactNode;
  className?: string;
}

interface AccordionContentProps
  extends React.ComponentPropsWithoutRef<typeof Accordion.Content> {
  children: React.ReactNode;
  className?: string;
}

export default function Faq() {
  const { t, i18n } = useTranslation();
  const [faqs, setFaqs] = useState<FaqData[]>([]);
  const [loading, setLoading] = useState<boolean>(true);
  const currentSelectedLanguage = localStorage.getItem("i18nextLng") || "en";

  const AccordionDemo = ({ faqs }: { faqs: FaqData[] }) => (
    <Accordion.Root
      className={styles.AccordionRoot}
      type="single"
      defaultValue={faqs[0]?.faq_id.toString()}
      collapsible
    >
      {faqs.map((faq) => (
        <Accordion.Item
          className={styles.AccordionItem}
          value={faq.faq_id.toString()}
          key={faq.faq_id}
        >
          <AccordionTrigger>
            {currentSelectedLanguage === "en"
              ? faq.question_en
              : faq.question_bm}
          </AccordionTrigger>
          <AccordionContent>
            {currentSelectedLanguage === "en" ? faq.answer_en : faq.answer_bm}
          </AccordionContent>
        </Accordion.Item>
      ))}
    </Accordion.Root>
  );

  const AccordionTrigger = React?.forwardRef<
    HTMLButtonElement,
    AccordionTriggerProps
  >(({ children, className, ...props }, forwardedRef) => (
    <Accordion.Header className={styles.AccordionHeader}>
      <Accordion.Trigger
        className={classNames(styles.AccordionTrigger, className)}
        {...props}
        ref={forwardedRef}
      >
        {children}
        <ChevronDownIcon className={styles.AccordionChevron} aria-hidden />
      </Accordion.Trigger>
    </Accordion.Header>
  ));

  const AccordionContent = React?.forwardRef<
    HTMLDivElement,
    AccordionContentProps
  >(({ children, className, ...props }, forwardedRef) => (
    <Accordion.Content
      className={classNames(styles.AccordionContent, className)}
      {...props}
      ref={forwardedRef}
    >
      <div className={styles.AccordionContentText}>{children}</div>
    </Accordion.Content>
  ));

  useEffect(() => {
    const getFaqs = async () => {
      try {
        // Perform the API call and destructure the `data` property from the response
        const { data } = await FetchFaq();
        // Set the FAQ data state with the returned data
        setFaqs(data);
      } catch (error) {
        console.error("Error fetching FAQs:", error);
      } finally {
        // Set loading to false regardless of the result
        setLoading(false);
      }
    };

    getFaqs();
  }, []);

  // if (loading) {
  //   return (
  //     <div className="loading_wrapper">
  //       <div className="loading_circle"></div>
  //       <div className="loading_circle"></div>
  //       <div className="loading_circle"></div>
  //       <div className="loading_shadow"></div>
  //       <div className="loading_shadow"></div>
  //       <div className="loading_shadow"></div>
  //       <span>Loading</span>
  //     </div>
  //   );
  // }

  return (
    <div>
      <img
        src="./images/faq.png"
        alt="Frequently Asked Questions"
        className={styles.faqImage}
      />
      <AccordionDemo faqs={faqs} />
    </div>
  );
}
