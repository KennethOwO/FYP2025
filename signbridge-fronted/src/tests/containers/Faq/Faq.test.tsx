import React from "react";
import Faq from "@root/containers/Faq/Faq";
import { create } from "react-test-renderer";
import { cleanup } from "@testing-library/react";
jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
})); // Mock AccordionTrigger
describe("Test Faq", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const tree = create(<Faq />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
