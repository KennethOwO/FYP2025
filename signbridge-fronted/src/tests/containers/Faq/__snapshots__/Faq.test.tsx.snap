// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Faq should render correctly 1`] = `
<p>
  loading_faq
</p>
`;
